lwt/cohttp_lwt.ml

```diff
@@ -194,21 +210,32 @@ module Make_client
   let callv ?(ctx=default_ctx) uri reqs =
     lwt (conn, ic, oc) = Net.connect_uri ~ctx uri in
     (* Serialise the requests out to the wire *)
-    let _ = Lwt_stream.iter_s (fun (req,body) ->
-        Request.write (fun writer ->
-            Cohttp_lwt_body.write_body (Request.write_body writer) body) req oc)
-        reqs in
-    (* Read the responses. For each response, ensure that the previous response
-     * has consumed the body before continuing to the next response, since HTTP/1.1
-     * pipelining cannot be interleaved. *)
+    lwt meths = Lwt_stream.fold_s (fun (req,body) meths ->
+      Request.write (fun writer ->
+        Cohttp_lwt_body.write_body (Request.write_body writer) body
+      ) req oc >>= fun () ->
+      return ((Request.meth req)::meths)
```

Comment:
Why do you need to fold over requests to get the methods. Can't you just Lwt_stream.map_s and return the method one by one?